views/find_missing.py

```diff
@@ -92,9 +114,136 @@ def render_search_button(self):
+    #     os.system(command)
+    
+    def update_init_search(self):
+        # print("Update Init Search")
         deck_name = self.deck_selection_enabled and self.deck_chooser.deckName()
-        model_fields = self.model_field_selection_enabled and self.model_field_chooser.selected_items
+        model_fields = self.model_field_selection_enabled and self.model_field_selected_items
+
+        if self.model_field_selection_enabled:
+            noteTypesSelected = [mod['name'] for mod in model_fields] if model_fields else ""
+            fieldsSelected = {fields['name'] for mod in model_fields for fields in mod['fields']} if model_fields else ""
+            print(noteTypesSelected, fieldsSelected)
 
+        self.initQuery = ""
+        self.initQuery += self.search_formatter(True, "deck", deck_name) if deck_name else ""
+        self.initQuery += self.search_multiple_terms_formatter("note", noteTypesSelected) if self.model_field_selection_enabled else ""
+        self.initQuery += self.search_multiple_fields_formatter(fieldsSelected, self.REPLACEMENT_STRING) if self.model_field_selection_enabled else ""
+        
+        self.search_preview.setText(self.get_final_search("[Word]"))
+
+
+
+    def search(self):
+        # Search through fetched cards here
+        
+        self.results_area.clear()
+        self.results.clear()
+        
+        self.update_init_search()
+
+        words = {self.text_area_text.toPlainText().split()}
+
+        for word in words:
+            query = self.get_final_search(word)
+        
+            results = mw.col.findCards(query)
+            hasResults = len(results) > 0
+            print(query + ": " + str(hasResults))
+
+            if not hasResults:
+                self.results.append(word)
+                self.results_area.addItem(self.create_result_item(word))
+                
+        print(self.results)
+        runHook('newMissingCardsResults', self.results)
+
+
+    def search_formatter(self, encapsulated, field, term, withSpace=True):
+        search = ""
+
+        if encapsulated:
+            search += "\""        
+
+        if field is False:
+            search += str(term)
+        else:
+            search += str(field) + ":" + str(term)
+        
+        if encapsulated:
+            search += "\""
+        
+        if withSpace:
+            search += " "
+        
+        return search
+
+    def search_multiple_terms_formatter(self, field, terms):
+        if not terms:
+            return ""
+        elif len(terms) == 1:
+            return self.search_formatter(True, field, terms[0])
+        
+        search = "("
+        search += " or ".join(self.search_formatter(True, field, x, False) for x in terms)
+        search += ") "
+
+        return search
+
+    def search_multiple_fields_formatter(self, fields, term):
+        if not fields:
+            return ""
+        elif len(fields) == 1:
+            return self.search_formatter(True, fields[0], term)
```

Comment:
Getting an type error here because `fields` is a set and is not subscriptable. We should wrap the set in `list()`. That is unless you meant for it to be a dictionary.

```
Caught exception:
  File "/home/dev/.local/share/Anki2/addons21/anki_LL/views/find_missing.py", line 100, in toggle_model_field_selection
    self.update_init_search()
  File "/home/dev/.local/share/Anki2/addons21/anki_LL/views/find_missing.py", line 164, in update_init_search
    self.initQuery += self.search_multiple_fields_formatter(fieldsSelected, self.REPLACEMENT_STRING) if self.model_field_selection_enabled else ""
  File "/home/dev/.local/share/Anki2/addons21/anki_LL/views/find_missing.py", line 230, in search_multiple_fields_formatter
    return self.search_formatter(True, fields[0], term)
<class 'TypeError'>: 'set' object is not subscriptable
```

Reply:
... I do not get this error. Just double checked, everything works on my end. Hmm, strange...

Reply:
Alright, just pushed a change here. I have rewrapped the set in `list( )` and it still functions for me. Let me know if this works for you :thinking: I wonder how we get different results.

Reply:
Forgot to add my reproduction instructions. In the note/field chooser, I selected none and then just selected one of the fields in a note type, but not all. After toggling off and on the note type filter I got the error.

Reply:
Nope, can't reproduce it. Going over this on video whenever you're free would be useful. Or just a screen recording yourself.

Reply:
It'll work now that the set has been wrapped in a list. Was just including how I originally got there. This is resolved.